internal/services/controllers/v1/olap/analyze.go

```diff
@@ -0,0 +1,28 @@
+package olap
+
+import (
+	"context"
+)
+
+func (oc *OLAPControllerImpl) runAnalyze(ctx context.Context) func() {
+	return func() {
+		oc.l.Debug().Msgf("analyze: running analyze on partitioned tables")
+
+		tenant, err := oc.p.GetInternalTenantForController(ctx)
```

Comment:
Note that we are moving away from using the internal tenant logic and instead favoring advisory locks on the database level, because the internal tenant risks duplicate operations running when the internal tenant moves between controllers.